tests/conformance/state/state.go

```diff
@@ -298,6 +298,45 @@ func ConformanceTests(t *testing.T, props map[string]string, statestore state.St
 		})
 	}
 
+	if config.HasOperation("get") && config.HasOperation("set") {
+		t.Run("read, write, read output match", func(t *testing.T) {
+			for _, scenario := range scenarios {
+				if !scenario.bulkOnly && !scenario.transactionOnly {
+					t.Logf("Retrieving value for %s", scenario.key)
+					req := &state.GetRequest{
+						Key: scenario.key,
+					}
+					if len(scenario.contentType) != 0 {
+						req.Metadata = map[string]string{metadata.ContentType: scenario.contentType}
+					}
+					res, err := statestore.Get(req)
+					assert.Nil(t, err)
+					assertEquals(t, scenario.value, res)
```

Comment:
The `assertEquals` method actually properly unmarshals, the `res.Data` field and compares it to `scenario.value` ...